docs/development.md

```diff
@@ -0,0 +1,182 @@
+* [mx4-commercial](https://github.com/hostmobility/mx4-commercial) - Source for drivers and tools for mx-4.
+
+## Building with Yocto
+
+## Bygg script 
+Since building with Yocto can be quite complex, we have [mobility-poky-platform](https://github.com/hostmobility/mobility-poky-platform.git) which automates much of the build tasks.
+
+This requires only installation of Docker before starting the build
+
+*Build example*
+```bash
+DL_DIR=/YOCTO_DOWNLOADS 
+scripts/bygg \ 
+  --manifest-file kirkstone \
+  --delete-conf \
+  --distro poky \
+  --machine imx8mp-var-dart-hmx1 \
+  bitbake console-hostmobility-image
+```
+
+Host Mobility Hardware run a Linux based system. As such, software is usually built from source. The target architechure is ARM32 and ARM64.
+
+## Host Mobility poky platform
+
+You can choose to build Yocto/Open Embedded the regular way or using our custom build-in-docker ("bid") script.
+
+## Regular Host Mobility Yocto/OE-core setup
+
+To simplify installation we provide a [repo](http://code.google.com/p/git-repo) manifest which manages the different git repositories
+and the versions used.
+
+Before proceeding, please take a look at [The Build Host Packages](http://www.yoctoproject.org/docs/2.3/mega-manual/mega-manual.html#packages) in the Yocto Manual.
+
+Install the repo bootstrap binary:
+
+```
+sudo apt-get install repo
+```
+
+or
+
+```
+mkdir ~/bin
+PATH=~/bin:$PATH
+curl http://commondatastorage.googleapis.com/git-repo-downloads/repo > ~/bin/repo
+chmod a+x ~/bin/repo
+```
+
+Create a directory for your `mobility-bsp-platform` setup to live in and clone the meta information.
+```
+mkdir mobility-bsp-platform
+cd mobility-bsp-platform
+repo init -u https://github.com/hostmobility/mobility-poky-platform -b master
+repo init -m kirkstone-next.xml
+repo sync --force-sync
+```
+
+Set up environment variables:
+```
+export DIR_WORK=$PWD/../
+export BUILD_TAG=*yourTag*
+export PLATFORM_VERSION="$(git -C $DIR_WORK.repo/manifests rev-parse --short HEAD)"
+export PLATFORM_VERSION_DETAILS="$(repo forall -c 'echo $REPO_PATH\nLREV: $REPO_LREV\nRREV: $REPO_RREV; git diff --stat -b $REPO_LREV..HEAD ; echo -n "Commit: " ; git rev-parse HEAD ; echo -n "Uncommited changes: " ; git status -b -s ; git diff --stat -b ; echo ')"
+echo "building with repo versions: $PLATFORM_VERSION"
+export BB_ENV_PASSTHROUGH_ADDITIONS="$BB_ENV_PASSTHROUGH_ADDITIONS BUILD_TAG PLATFORM_VERSION PLATFORM_VERSION_DETAILS"
+export TEMPLATECONF=$PWD/../sources/meta-mobility-poky-distro/conf
+```
+**NOTE!** You will need to perform these exports for each new session. If you already have
+a build directory, it will be untouched and only the environment variables will be set.
+
+Start baking!
+```
+$ source sources/poky/oe-init-build-env build;
+$ bitbake {image}
+```
+
+where `{image}` is replaced by, for example, `console-mobility-image`, `bitbake mobility-image` or `mobility-image-chromium`.
```

Comment:
there is a list of supported images when it is sourced. and which machine that is supported maybe link to those files? hmx(https://github.com/hostmobility/meta-mobility-poky-distro/blob/master/conf/hmx/conf-notes.txt)